egs2/libriheavy_small/asr2/local/run_hf.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+
+kmeans_feature="wavlm_large/21"  # use model_type/layer_index
+nclusters=2000
+
+src_lang=$(echo "${kmeans_feature}_km${nclusters}" | tr "/" "_")
+tgt_lang=en
+
+train_set="train_small"
+train_dev="dev"
+test_sets="dev test_clean test_other"
+
+asr_config_ctc=conf/tuning/train_discrete_asr_e_branchformer_mistral02_ctc.yaml
+asr_config_aed=conf/tuning/train_discrete_asr_e_branchformer_mistral02_aed.yaml
+inference_config=conf/decode_hf.yaml
+
+src_nbpe=6000   # I use src_nbpe=6000 for 2000-cluster kmeans.
+tgt_nbpe=3000   # if token_joint is True, then only tgt_nbpe is used
+
+# ts: true sequence
+# rm: deduplicated sequence which removes duplicated tokens
+src_case="rm"
+tgt_case="ts"
+
+embed_tokens_path=data/mistral7b-instruct02_embed_tokens.pth
+
+local/export_hf_embed_tokens.sh \
+    mistralai/Mistral-7B-Instruct-v0.2 \
+    ${embed_tokens_path}
+
+./asr2.sh \
```

Comment:
Can you add some comments why running asr2.sh twice?

Reply:
Done! Please let me know if I should clarify it more.